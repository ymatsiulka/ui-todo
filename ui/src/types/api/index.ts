export * from './TodoResponse';
export * from './TodoRequest';
