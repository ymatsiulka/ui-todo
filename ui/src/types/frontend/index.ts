export * from './props/SpacerProps';
export * from './props/ButtonProps';
export * from './props/TodoPageStatusesProps';
