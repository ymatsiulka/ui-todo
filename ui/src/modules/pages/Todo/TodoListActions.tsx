import React from 'react';
import { TodoPageStatuses } from '../../../types/frontend';
import Button from '../../../modules/shared/button/Button';

interface TodoListActionsProps {
  todoPageStatus: TodoPageStatuses;
  uncompletedItemsCount: number;
  onClickAllHandler: React.MouseEventHandler<HTMLButtonElement>;
  onClickActiveHandler: React.MouseEventHandler<HTMLButtonElement>;
  onClickCompletedHandler: React.MouseEventHandler<HTMLButtonElement>;
  onClickClearCompletedHandler: React.MouseEventHandler<HTMLButtonElement>;
}

const TodoListActions: React.FC<TodoListActionsProps> = ({
  todoPageStatus,
  uncompletedItemsCount,
  onClickAllHandler,
  onClickActiveHandler,
  onClickCompletedHandler,
  onClickClearCompletedHandler,
}) => {
  const isActiveButton = (isActive: boolean) => `fs-12 ${isActive && 'c-blue'}`;

  return (
    <div className="todo-list-actions">
      <div className="left">
        <span className="fs-12">{uncompletedItemsCount} items left </span>
      </div>
      <div className="middle">
        <Button
          className={isActiveButton(todoPageStatus == TodoPageStatuses.All)}
          title="All"
          onClick={onClickAllHandler}
        />
        <Button
          className={isActiveButton(todoPageStatus == TodoPageStatuses.Active)}
          title="Active"
          onClick={onClickActiveHandler}
        />
        <Button
          className={isActiveButton(todoPageStatus == TodoPageStatuses.Completed)}
          title="Completed"
          onClick={onClickCompletedHandler}
        />
      </div>
      <div className="right">
        <Button className="fs-12" title="Clear Completed" onClick={onClickClearCompletedHandler} />
      </div>
    </div>
  );
};

export default TodoListActions;
