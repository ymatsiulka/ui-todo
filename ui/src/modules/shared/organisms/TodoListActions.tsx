import React from 'react';
import { TodoPageStatuses } from 'types/frontend';
import { Button } from 'modules/shared/atoms';
import styles from './TodoListActions.m.scss';

interface TodoListActionsProps {
  todoPageStatus: TodoPageStatuses;
  uncompletedItemsCount: number;
  onClickAllHandler: React.MouseEventHandler<HTMLButtonElement>;
  onClickActiveHandler: React.MouseEventHandler<HTMLButtonElement>;
  onClickCompletedHandler: React.MouseEventHandler<HTMLButtonElement>;
  onClickClearCompletedHandler: React.MouseEventHandler<HTMLButtonElement>;
}

const TodoListActions: React.FC<TodoListActionsProps> = ({
  todoPageStatus,
  uncompletedItemsCount,
  onClickAllHandler,
  onClickActiveHandler,
  onClickCompletedHandler,
  onClickClearCompletedHandler,
}) => {
  const activeButtonClassName = (isActive: boolean): string => `fs-12 ${isActive && 'c-blue'}`;

  return (
    <div className={styles.todoListActions}>
      <div className="left">
        <span className="fs-12">{uncompletedItemsCount} items left </span>
      </div>
      <div className="middle">
        <Button
          id="show-all-todo-btn"
          className={activeButtonClassName(todoPageStatus === TodoPageStatuses.All)}
          title="All"
          onClick={onClickAllHandler}
        />
        <Button
          id="show-active-todo-btn"
          className={activeButtonClassName(todoPageStatus === TodoPageStatuses.Active)}
          title="Active"
          onClick={onClickActiveHandler}
        />
        <Button
          id="show-completed-todo-btn"
          className={activeButtonClassName(todoPageStatus === TodoPageStatuses.Completed)}
          title="Completed"
          onClick={onClickCompletedHandler}
        />
      </div>
      <div className="right">
        <Button className="fs-12" title="Clear Completed" onClick={onClickClearCompletedHandler} />
      </div>
    </div>
  );
};

export default TodoListActions;
