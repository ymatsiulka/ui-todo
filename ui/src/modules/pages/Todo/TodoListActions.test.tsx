import React from 'react';
import { render } from '@testing-library/react';
import TodoListActions from './TodoListActions';
import { TodoPageStatuses } from 'types/frontend';

jest.mock('modules/shared', () => ({
  Button: jest.fn((props) => <button {...props} />),
}));

test('TodoListActions should render properly', async () => {
  // Arrange
  // Act
  const { asFragment } = render(
    <TodoListActions
      uncompletedItemsCount={1}
      todoPageStatus={TodoPageStatuses.Active}
      onClickAllHandler={jest.fn()}
      onClickActiveHandler={jest.fn()}
      onClickCompletedHandler={jest.fn()}
      onClickClearCompletedHandler={jest.fn()}
    />,
  );

  // Assert
  expect(asFragment()).toMatchSnapshot();
});
