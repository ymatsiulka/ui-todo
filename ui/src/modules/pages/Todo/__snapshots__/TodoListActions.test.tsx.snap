// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly 1`] = `
<DocumentFragment>
  <div
    class="todoListActions"
  >
    <div
      class="left"
    >
      <span
        class="fs-12"
      >
        1 items left 
      </span>
    </div>
    <div
      class="middle"
    >
      <button
        class="button fs-12 false"
        type="button"
      >
        <span>
          All
        </span>
      </button>
      <button
        class="button fs-12 c-blue"
        type="button"
      >
        <span>
          Active
        </span>
      </button>
      <button
        class="button fs-12 false"
        type="button"
      >
        <span>
          Completed
        </span>
      </button>
    </div>
    <div
      class="right"
    >
      <button
        class="button fs-12"
        type="button"
      >
        <span>
          Clear Completed
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TodoListActions should render properly 1`] = `
<DocumentFragment>
  <div
    class="todoListActions"
  >
    <div
      class="left"
    >
      <span
        class="fs-12"
      >
        1 items left 
      </span>
    </div>
    <div
      class="middle"
    >
      <button
        class="button fs-12 false"
        type="button"
      >
        <span>
          All
        </span>
      </button>
      <button
        class="button fs-12 c-blue"
        type="button"
      >
        <span>
          Active
        </span>
      </button>
      <button
        class="button fs-12 false"
        type="button"
      >
        <span>
          Completed
        </span>
      </button>
    </div>
    <div
      class="right"
    >
      <button
        class="button fs-12"
        type="button"
      >
        <span>
          Clear Completed
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
